### PR TITLE
Add support for sectionHeader field in Select dropdown

### PR DESCRIPTION
**Pull request by heysweet**
This PR is needed for [dbt-cloud/PR#2516](https://github.com/fishtown-analytics/dbt-cloud/pull/2516).

Add an optional header string for each section in a Select dropdown.

image

### CODE REVIEW

File: components/Forms/Select/Select.tsx [hunk 1]
```diff
@@ -8,7 +8,8 @@ import { Icon } from '../../Icon';
 
 export interface SelectOption {
   label: string;
-  value?: string;
+  sectionHeader?: string;
+  value?: any;
```
Comment (heysweet) [inline components/Forms/Select/Select.tsx:12]: This change was needed to have parity with the FilterSelect use-case in my PR.

File: components/Forms/Select/select.css [hunk 2]
```diff
@@ -47,6 +47,11 @@
   @apply tw-flex tw-align-middle tw-justify-between;
 }
 
+.fui-select-container .fui-select__section-header {
+  font-size: 15px;
```
Comment (jgiuffrida) [inline components/Forms/Select/select.css:51]: Is there a way to do with tailwind `@apply` rules?

**Review by jgiuffrida — COMMENTED** [review]
One tiny nit - other than that it looks good

File: components/Forms/Select/select.css (hunk 2 above)
Comment (heysweet) [reply components/Forms/Select/select.css:51]: Probably! Have never used tailwind, will try to look for the correct way to do this.

**Pull request merged**